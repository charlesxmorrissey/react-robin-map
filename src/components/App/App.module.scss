.app {
  height: 100vh;
  position: relative;
  width: 100vw;
}

.appMain {
  height: 100vh;
  overflow: hidden;
  position: fixed;
  right: 0;
  top: 0;
  width: calc(100vw - 300px);

  @media screen and (max-width: 65.625em), screen and (max-height: 40.625em) {
    width: 100vw;
  }
}

.appStage {
  height: 100%;
  pointer-events: none;
  position: relative;
  transition-timing-function: cubic-bezier(0.2, 1, 0.3, 1);
  transition: transform var(--duration-medium-transition);
  width: 100%;
}

.appStageContentOpen {
  transform: translate3d(0, -25%, 0) scale3d(0.8, 0.8, 1);
}
