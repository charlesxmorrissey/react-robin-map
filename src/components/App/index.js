import React, { useEffect, useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useLocation, useHistory } from 'react-router-dom'

import {
  fetchLocations,
  fetchSpaces,
  fetchSpace,
  resetSpace,
} from 'store/robin/actions'
import { selectCurrentLocation, selectShowSidebar } from 'store/selectors'
import { getLocationFromPathname, getSpaceIdFromPathname } from 'utils/robin'
import { LOCATIONS, LOCATION_DEFAULT } from 'constants/index'

import Levels from 'components/Levels'
import Sidebar from 'components/Sidebar'

import styles from './App.module.scss'

const App = () => {
  const dispatch = useDispatch()
  const location = useLocation()
  const history = useHistory()
  const currentLocation = useSelector(selectCurrentLocation())
  const showSidebar = useSelector(selectShowSidebar())

  useEffect(() => {
    dispatch(fetchLocations())
  }, [dispatch])

  // If location is anything other than a valid office, route to default
  const onPathnameUpdate = useCallback(() => {
    const locationName = getLocationFromPathname(location.pathname)
    const spaceId = getSpaceIdFromPathname(location.pathname)

    if (locationName) {
      dispatch(fetchSpaces(locationName))
    } else {
      history.push(LOCATION_DEFAULT)
    }

    if (spaceId) {
      dispatch(fetchSpace(spaceId))
    } else {
      dispatch(resetSpace())
    }
  }, [location, dispatch, history])

  useEffect(() => {
    onPathnameUpdate()
  }, [onPathnameUpdate])

  return (
    <main className={styles.app}>
      <div className={styles.appMain}>
        <div className={styles.appStage}>
          <Levels
            locationKey={
              currentLocation
                ? currentLocation.name
                : LOCATIONS[LOCATION_DEFAULT]
            }
          />
        </div>
      </div>

      <Sidebar showSidebar={showSidebar} />
    </main>
  )
}

export default App
