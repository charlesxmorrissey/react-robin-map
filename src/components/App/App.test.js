import React from 'react'
import { Router } from 'react-router-dom'

import Adapter from 'enzyme-adapter-react-16'
import { mount, configure } from 'enzyme'
import { createMemoryHistory } from 'history'

import App from './index'
// import Sidebar from '../Sidebar'

configure({ adapter: new Adapter() })

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}))

describe('<App />', () => {
  it('should match the snapshot', () => {
    const history = createMemoryHistory({ initialEntries: ['/bk/51337'] })
    const component = mount(
      <Router history={history}>
        <App />
      </Router>
    )
    expect(component.find(App)).toMatchSnapshot()
  })

  it('should match the snapshot', () => {
    const history = createMemoryHistory()
    const component = mount(
      <Router history={history}>
        <App />
      </Router>
    )
    expect(component.find(App)).toMatchSnapshot()
  })
})
