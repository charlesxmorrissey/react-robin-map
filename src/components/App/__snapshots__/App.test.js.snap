// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
<App>
  <main
    className="app"
  >
    <div
      className="appMain"
    >
      <div
        className="appStage"
      >
        <Levels
          location="/bk/51337"
        >
          <div
            className="streetLevel"
          >
            <ForwardRef(SvgStreetLevel)
              className="streetLevelMap"
            >
              <svg
                className="streetLevelMap"
              >
                street-level.svg
              </svg>
            </ForwardRef(SvgStreetLevel)>
          </div>
          <div
            className="levels"
          >
            <LevelItem>
              <div
                className="levelItem"
              >
                <ForwardRef(SvgLevel1)>
                  <svg>
                    level-1.svg
                  </svg>
                </ForwardRef(SvgLevel1)>
              </div>
            </LevelItem>
            <LevelItem>
              <div
                className="levelItem"
              >
                <ForwardRef(SvgLevel2)>
                  <svg>
                    level-2.svg
                  </svg>
                </ForwardRef(SvgLevel2)>
              </div>
            </LevelItem>
          </div>
        </Levels>
      </div>
    </div>
    <Sidebar>
      <aside
        className="sidebar"
      >
        <Search>
          <div
            className="search"
          >
            <input
              className="searchInput"
              placeholder="Search..."
            />
            <button
              aria-label="Close Search"
              className="searchCloseBtn"
            >
              <ForwardRef(SvgClose)
                className="searchCloseBtnIcon"
              >
                <svg
                  className="searchCloseBtnIcon"
                >
                  close.svg
                </svg>
              </ForwardRef(SvgClose)>
            </button>
          </div>
        </Search>
      </aside>
    </Sidebar>
  </main>
</App>
`;

exports[`<App /> should match the snapshot 2`] = `
<App>
  <main
    className="app"
  >
    <div
      className="appMain"
    >
      <div
        className="appStage"
      >
        <Levels
          location="/bk"
        >
          <div
            className="streetLevel"
          >
            <ForwardRef(SvgStreetLevel)
              className="streetLevelMap"
            >
              <svg
                className="streetLevelMap"
              >
                street-level.svg
              </svg>
            </ForwardRef(SvgStreetLevel)>
          </div>
          <div
            className="levels"
          >
            <LevelItem>
              <div
                className="levelItem"
              >
                <ForwardRef(SvgLevel1)>
                  <svg>
                    level-1.svg
                  </svg>
                </ForwardRef(SvgLevel1)>
              </div>
            </LevelItem>
            <LevelItem>
              <div
                className="levelItem"
              >
                <ForwardRef(SvgLevel2)>
                  <svg>
                    level-2.svg
                  </svg>
                </ForwardRef(SvgLevel2)>
              </div>
            </LevelItem>
          </div>
        </Levels>
      </div>
    </div>
    <Sidebar>
      <aside
        className="sidebar"
      >
        <Search>
          <div
            className="search"
          >
            <input
              className="searchInput"
              placeholder="Search..."
            />
            <button
              aria-label="Close Search"
              className="searchCloseBtn"
            >
              <ForwardRef(SvgClose)
                className="searchCloseBtnIcon"
              >
                <svg
                  className="searchCloseBtnIcon"
                >
                  close.svg
                </svg>
              </ForwardRef(SvgClose)>
            </button>
          </div>
        </Search>
      </aside>
    </Sidebar>
  </main>
</App>
`;
